compiler/IREmitter/Payload/codegen-payload.c

```diff
@@ -704,6 +704,25 @@ VALUE sorbet_int_hash_to_h(VALUE recv, ID fun, int argc, const VALUE *const rest
     return sorbet_rb_hash_to_h(recv);
 }
 
+SORBET_INLINE
+VALUE sorbet_rb_hash_fetch_m_withBlock(VALUE recv, ID fun, int argc, const VALUE *const restrict argv, BlockFFIType blk,
+                                       const struct rb_captured_block *captured, VALUE closure, int numPositionalArgs) {
+    extern VALUE sorbet_hash_stlike_lookup(VALUE hash, VALUE key, VALUE * val);
+
+    rb_check_arity(argc, 1, 2);
```

Comment:
iirc* when argc==2 the interpreter will issue a warning that the second arg is silently dropped (because the block overrides it)---I think skipping the warning is probably fine, but you might add a comment about why we're checking for (1,2) then never looking at argv[1].

*iidnrc then please disregard :)

Reply:
I figured that it would be okay to suppress the warning here, but you're right that a comment explaining that would be nice.